be/src/vec/common/custom_allocator.h

```diff
@@ -33,6 +35,69 @@ template <class Key, class T, class Compare = std::less<Key>,
           class Allocator = CustomStdAllocator<std::pair<const Key, T>>>
 using DorisMap = std::map<Key, T, Compare, Allocator>;
 
+template <typename T>
+struct DorisUniqueBufferDeleter {
+    Allocator<false> allocator;
```

Comment:
`Allocator` is an empty class. maybe inheriting is a better solution than being a member because of EBO(empty base class optimization)